src/mlpack/core/optimizers/svrg/barzilai_borwein.hpp

```diff
@@ -0,0 +1,109 @@
+/**
+ * @file barzilai_borwein.hpp
+ * @author Marcus Edel
+ *
+ * Barzilai-Borwein decay policy.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_CORE_OPTIMIZERS_SVRG_BARZILIA_BORWEIN_HPP
+#define MLPACK_CORE_OPTIMIZERS_SVRG_BARZILIA_BORWEIN_HPP
+
+#include <mlpack/prereqs.hpp>
+
+namespace mlpack {
+namespace optimization {
+
+/**
+ * Barzilai-Borwein decay policy for Stochastic variance reduced gradient
+ * (SVRG).
+ *
+ * For more information, please refer to:
+ *
+ * @code
+ * @incollection{Tan2016,
+ *   title     = {Barzilai-Borwein Step Size for Stochastic Gradient Descent},
+ *   author    = {Tan, Conghui and Ma, Shiqian and Dai, Yu-Hong
+ *                and Qian, Yuqiu},
+ *   booktitle = {Advances in Neural Information Processing Systems 29},
+ *   editor    = {D. D. Lee and M. Sugiyama and U. V. Luxburg and I. Guyon
+ *                and R. Garnett},
+ *   pages     = {685--693},
+ *   year      = {2016},
+ *   publisher = {Curran Associates, Inc.}
+ * }
+ * @endcode
+ */
+class BarzilaiBorwein
```

Comment:
Do you think we should call it `BarzilaiBorweinDecay` for clarity?  I'm fine with either way, just a suggestion.